### PR TITLE
Pass frequencies arrays of same length to VAFPlot

### PR DESCRIPTION
**Pull request by adamabeshouse**
See https://github.com/cBioPortal/cbioportal/issues/2361 for discussion

# Checks
- [x] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [x] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [x] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

### CODE REVIEW

File: src/pages/patientView/genomicOverview/GenomicOverview.spec.tsx [hunk 1]
```diff
@@ -1,21 +1,182 @@
...
+                sampleOrder={{}}
+                sampleLabels={{}}
+                sampleColors={{}}
+                sampleManager={new SampleManager([])}
+                getContainerWidth={()=>20}
+            />);
+            frequencies = getVAFFrequencies(GO);
+            assertDeepEqualFrequencies(frequencies, {sample1: [1/4, 1/6, 1/5, 1/11]}, "correctly calculates, part 2");
+        });
+        it("computes the correct frequencies for multiple samples, such that "+
```
Comment (alisman) [inline src/pages/patientView/genomicOverview/GenomicOverview.spec.tsx]: don't forget the joy of using multi-line strings with ``

File: src/pages/patientView/genomicOverview/GenomicOverview.tsx [hunk 2]
```diff
@@ -77,22 +74,31 @@ export default class GenomicOverview extends React.Component<IGenomicOverviewPro
...
-        const ret:{[s:string]:number[]} = {};
+    private computeMutationFrequencyBySample(mergedMutations:Mutation[][]):MutationFrequenciesBySample {
+        const ret:MutationFrequenciesBySample = {};
         let sampleId;
         let freq;
-        for (let mutation of mutations) {
-            sampleId = mutation.sampleId;
-            freq = mutation.tumorAltCount / (mutation.tumorRefCount + mutation.tumorAltCount);
+        for (let mutations of mergedMutations) {
+            for (const mutation of mutations) {
```
Comment (onursumer) [inline src/pages/patientView/genomicOverview/GenomicOverview.tsx]: why `let mutations` but `const mutation`? Is there a typescript error if you do `const mutations`?

File: src/pages/patientView/genomicOverview/GenomicOverview.spec.tsx [hunk 3]
```diff
@@ -1,21 +1,182 @@
...
-import { shallow, mount } from 'enzyme';
+import {shallow, mount, ShallowWrapper, ReactWrapper} from 'enzyme';
 import sinon from 'sinon';
+import SampleManager from "../sampleManager";
+import {ThumbnailExpandVAFPlot} from "../vafPlot/ThumbnailExpandVAFPlot";
+import {Mutation} from "../../../shared/api/generated/CBioPortalAPI";
+import {MutationFrequenciesBySample} from "../vafPlot/VAFPlot";
 
 describe('GenomicOverview', () => {
+    let GO:ShallowWrapper<any,any>;
```
Comment (onursumer) [inline src/pages/patientView/genomicOverview/GenomicOverview.spec.tsx]: Can we rename this to `genomicOverview` or something similar? It looks like a constant to me when it is all uppercase :)

File: src/pages/patientView/genomicOverview/GenomicOverview.tsx (hunk 2 above)
Comment (adamabeshouse) [inline src/pages/patientView/genomicOverview/GenomicOverview.tsx]: hm, typo

**Pull request merged**